dvc/remote/local.py

```diff
@@ -681,16 +691,30 @@ def _process(
             return 0
 
         futures = []
-        with ThreadPoolExecutor(max_workers=jobs) as executor:
-            for from_infos, to_infos, names in chunks:
-                res = executor.submit(func, from_infos, to_infos, names=names)
-                futures.append(res)
-
+        tmp_files = []
+        try:
+            with ThreadPoolExecutor(max_workers=jobs) as executor:
+                for from_infos, to_infos, tmp_infos, names in chunks:
+                    tmp_files.extend(tmp_infos)
+                    res = executor.submit(func, from_infos, to_infos, tmp_infos, names=names)
+                    futures.append(res)
+        except (KeyboardInterrupt, Exception):
+            logger.error("Cleaning up temporary files...")
+            self._cleanup_tmp_files(remote, tmp_files)
+            raise
         for f in futures:
             f.result()
 
         return len(chunks)
 
+    def _cleanup_tmp_files(self, remote, tmp_files):
+        for i_info in tmp_infos:
+            logger.error("About to remove file {}".format(i_info))
+            if i_info["scheme"] == "local":
```

Comment:
The distinction should actually be done not based on scheme, but rather on whether or not we are pushing or pulling. For pulling/fetching it should be `self.remove(i_info)` and for pushing it should be `remote.remove(i_info)`